### PR TITLE
Shuffle by record rather than conformer

### PR DESCRIPTION
**Pull request by ArnNag**
## Description
Address #39 

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by ArnNag** [comment]
Do you know why it isn't running the checks @chrisiacovella @wiederm ?

File: modelforge/dataset/dataset.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 from modelforge.utils.prop import PropertyNames
 
-from modelforge.dataset.utils import RandomSplittingStrategy, SplittingStrategy
+from modelforge.dataset.utils import RandomRecordSplittingStrategy, SplittingStrategy
 
 
-class TorchDataset(torch.utils.data.Dataset):
+class TorchDataset(torch.utils.data.Dataset[Dict[str, torch.Tensor]]):
```
Comment (wiederm) [inline modelforge/dataset/dataset.py:15]: I didn't realize that this type hint was passible! Thanks!

File: modelforge/dataset/utils.py [hunk 2]
```diff
@@ -126,6 +126,161 @@ def split(self, dataset: "TorchDataset") -> Tuple[Subset, Subset, Subset]:
...
+        # add 1 to all the lengths in round-robin fashion until the remainder is 0
+        for i in range(remainder):
+            idx_to_add_at = i % len(subset_lengths)
+            subset_lengths[idx_to_add_at] += 1
+
+        lengths = subset_lengths
+
+        for i, length in enumerate(lengths):
+            if length == 0:
+                warnings.warn(f"Length of split at index {i} is 0. "
```
Comment (wiederm) [inline modelforge/dataset/utils.py:262]: can this really be the desired behavior?

**Review by wiederm — APPROVED** [review]

File: modelforge/dataset/utils.py (hunk 2 above)
Comment (ArnNag) [reply modelforge/dataset/utils.py:262]: This is copied from https://github.com/pytorch/pytorch/blob/5cfe973bed36099be7d7ddfd62136025430f8324/torch/utils/data/dataset.py#L444
